packages/commerce-sdk-react/src/hooks/utils.ts

```diff
@@ -1,86 +1,64 @@
 
-//TODO: update data type for updater when needed
-export interface QueryKeyMap {
-    name: string
-    key: QueryKey
-    updater?: Updater<
-        DataType<Client[CombinedMutationTypes]> | undefined,
-        DataType<Client[CombinedMutationTypes]> | undefined
-    >
-}
-
-export interface CacheUpdateMatrixElement {
-    update?: Array<QueryKeyMap>
-    invalidate?: Array<QueryKeyMap>
-    remove?: Array<QueryKeyMap>
-}
-
-// TODO: Add more endpoints types as needed
-export type CombinedMutationTypes =
-    | ShopperOrdersMutationType
-    | ShopperCustomersMutationType
-    | ShopperBasketsMutationType
-
-type CacheUpdateMatrix = {
-    [key in CombinedMutationTypes]?: (data: any, param: any) => CacheUpdateMatrixElement
+const deepEqual = (a: unknown, b: unknown): boolean => {
+    if (!isObject(a) || !isObject(b)) return a === b
+    const aEntries = Object.entries(a)
+    const bEntries = Object.entries(b)
+    if (aEntries.length !== bEntries.length) return false
+    return aEntries.every(([aKey, aValue], index) => {
+        const [bKey, bValue] = bEntries[index]
+        return aKey === bKey && deepEqual(aValue, bValue)
+    })
 }
 
-// TODO: Add more api clients as needed
-export type Client = ApiClients['shopperOrders'] &
-    ApiClients['shopperCustomers'] &
-    ApiClients['shopperBaskets']
+/**
+ * Determines whether the cache query key starts with the same values as the lookup query key
+ * @param cacheQueryKey query key from the cache
+ * @param lookupQueryKey partial query key to validate match against
+ * @returns boolean
+ */
+const isMatchingKey = (cacheQueryKey: QueryKey, lookupQueryKey: QueryKey): boolean =>
+    lookupQueryKey.every((item, index) => deepEqual(item, cacheQueryKey[index]))
 
-export const updateCache = <Action extends CombinedMutationTypes>(
+export const updateCache = (
     queryClient: QueryClient,
-    action: Action,
-    cacheUpdateMatrix: CacheUpdateMatrix,
-    response: DataType<Client[Action]>,
-    params: Argument<Client[Action]>
+    cacheUpdates: CacheUpdate,
+    response: unknown
 ) => {
-    const isMatchingKey = (cacheQuery: {queryKey: {[x: string]: any}}, queryKey: QueryKey) =>
-        queryKey.every((item, index) =>
-            isObject(item) && isObject(cacheQuery.queryKey[index])
-                ? Object.entries(cacheQuery.queryKey[index]).sort().toString() ===
-                  Object.entries(item as Record<string, unknown>)
-                      .sort()
-                      .toString()
-                : item === cacheQuery.queryKey[index]
-        )
-
-    // STEP 1. Update data inside query cache for the matching queryKeys, and updater
-    cacheUpdateMatrix[action]?.(params, response)?.update?.map(({key: queryKey, updater}) => {
-        queryClient.setQueryData(queryKey, updater)
+    // STEP 1. Update data inside query cache for the matching queryKeys
+    cacheUpdates.update?.forEach((queryKey) => {
+        queryClient.setQueryData(queryKey, response)
```

Comment:
Note: we've updated the signature for setQueryData to use a function `updater` because we want to update manually the cache on some mutations (e.g basket mutation) to